static/app/views/performance/newTraceDetails/traceLevelOpsBreakdown.tsx

```diff
@@ -0,0 +1,132 @@
+import styled from '@emotion/styled';
+
+import {addErrorMessage} from 'sentry/actionCreators/indicator';
+import {pickBarColor} from 'sentry/components/performance/waterfall/utils';
+import Placeholder from 'sentry/components/placeholder';
+import {IconCircleFill} from 'sentry/icons/iconCircleFill';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {NewQuery} from 'sentry/types/organization';
+import {useDiscoverQuery} from 'sentry/utils/discover/discoverQuery';
+import {DiscoverDatasets} from 'sentry/utils/discover/types';
+import type {Color} from 'sentry/utils/theme';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+
+import {useTraceEventView} from './useTraceEventView';
+import {type TraceViewQueryParams, useTraceQueryParams} from './useTraceQueryParams';
+
+function useTraceLevelOpsQuery(
+  traceSlug: string,
+  params: TraceViewQueryParams,
+  partialSavedQuery: Partial<NewQuery>
+) {
+  const location = useLocation();
+  const organization = useOrganization();
+  const eventView = useTraceEventView(traceSlug, params, {
+    ...partialSavedQuery,
+    dataset: DiscoverDatasets.SPANS_EAP,
+  });
+
+  return useDiscoverQuery({
+    eventView,
+    orgSlug: organization.slug,
+    location,
+  });
+}
+
+function LoadingPlaceHolder() {
+  return (
+    <Container>
+      <StyledPlaceholder _height={16} _width={100} />
+      <StyledPlaceholder _height={16} _width={100} />
+      <StyledPlaceholder _height={16} _width={100} />
+    </Container>
+  );
+}
+
+type Props = {
+  isTraceLoading: boolean;
+  traceSlug: string;
+};
+
+function TraceLevelOpsBreakdown({traceSlug, isTraceLoading}: Props) {
+  const urlParams = useTraceQueryParams();
+  const {
+    data: opsCountsResult,
+    isPending: isOpsCountsLoading,
+    isError: isOpsCountsError,
+  } = useTraceLevelOpsQuery(traceSlug ?? '', urlParams, {
+    fields: ['span.op', 'count()'],
+    orderby: '-count',
+  });
+  const {
+    data: totalCountResult,
+    isPending: isTotalCountLoading,
+    isError: isTotalCountError,
+  } = useTraceLevelOpsQuery(traceSlug ?? '', urlParams, {
+    fields: ['count()'],
+  });
+
+  if (isOpsCountsLoading || isTotalCountLoading || isTraceLoading) {
+    return <LoadingPlaceHolder />;
+  }
+
+  if (isOpsCountsError || isTotalCountError) {
+    addErrorMessage(t('Failed to load trace level ops breakdown'));
+    return null;
+  }
+
+  const totalCount = totalCountResult?.data[0]?.['count()'] ?? 0;
+
+  if (typeof totalCount !== 'number' || totalCount <= 0) {
+    return null;
+  }
+
+  return (
+    <Container>
+      {opsCountsResult?.data.slice(0, 4).map(currOp => {
```

Comment:
Why only first 4?

Reply:
Just followed the designs here for it: https://www.figma.com/design/zfReOSQiHfODLLoFWpYVdW/Trace-V3?node-id=42-1924&t=tsKe9q2Uvr3vObN0-0